src/scheduler/pbs_sched.c

```diff
@@ -1273,3 +1188,140 @@ main(int argc, char *argv[])
 	log_close(1);
 	exit(0);
 }
+
+/**
+ *
+ * @brief
+ *		Compares two sched commands(command in cache vs new command coming from socket) and finds out
+ *	        whether they are identical or not. Also replaces command in cache with the new command if new command
+ *	        is superset.
+ *	        Since this is a separate function more filtering criteria for Sched commands can be easily added. For example
+ *              if a new sched command is introduced or priority of an existing command has changed this function has to be
+ *              revisited.
+ *
+ * @param[in]	cmd_cache	-	command in Sched's cache
+ * @param[in]	new_cmd		-	command from socket buffer
+ *
+ *
+ * @return	int
+ * @retval	1	-	if both command are duplicate
+ * @reval	0	-	if not duplicate
+ *
+ */
+static int
+compare_modify_sched_cmd(sched_cmd *cmd_cache, sched_cmd *new_cmd)
+{
+	if (new_cmd->cmd == SCH_SCHEDULE_RESTART_CYCLE )
+		return 1;
+	
+	if (new_cmd->cmd == SCH_SCHEDULE_AJOB &&
+		(cmd_cache->cmd != SCH_SCHEDULE_AJOB && cmd_cache->cmd != SCH_CONFIGURE &&
+		 cmd_cache->cmd != SCH_RULESET && cmd_cache->cmd != SCH_ERROR)) {
+		return 1;
+	} else if (cmd_cache->cmd == SCH_SCHEDULE_AJOB &&
+		(new_cmd->cmd != SCH_SCHEDULE_AJOB && new_cmd->cmd != SCH_CONFIGURE &&
+		 new_cmd->cmd != SCH_RULESET && new_cmd->cmd != SCH_ERROR)) {
+		cmd_cache->cmd = new_cmd->cmd;
+		cmd_cache->jid = NULL;	
+		return 1;	
+	}
+
+	if (cmd_cache && new_cmd) {
+		if (cmd_cache->jid == NULL && new_cmd->jid == NULL) {
+			if (cmd_cache->cmd == new_cmd->cmd)
+				return 1;
+		} else {
+			if (cmd_cache->cmd == new_cmd->cmd && !strcmp(cmd_cache->jid, new_cmd->jid))
+				return 1;
+		}
+	}
+	return 0;
+}
+
+/**
+ * @brief Filters scheduler commands queue according to the criteria given in the function
+ *              compare_modify_sched_cmd
+ *
+* @param[in] sched_cmds	- scheduler command queue
+ *
+ * @return	ds_queue *
+ * @retval	success	- returns the filtered queue
+ * @reval	 fail -	NULL
+ *
+ */
+ds_queue *
+filter_sched_cmds(ds_queue *sched_cmds)
```

Comment:
Can you please add some comments to this function? It isn't easy to follow.

Reply:
I have added more comments in the code along with the algorithm. Please check my latest commit.